include/server/server.hpp

```diff
@@ -53,8 +52,7 @@ class Server
         const auto port_string = std::to_string(port);
 
         boost::asio::ip::tcp::resolver resolver(io_context);
-        boost::asio::ip::tcp::resolver::query query(address, port_string);
-        boost::asio::ip::tcp::endpoint endpoint = *resolver.resolve(query);
+        boost::asio::ip::tcp::endpoint endpoint = *resolver.resolve(address, port_string).begin();
```

Comment:
This change looks totally unrelated with the PR scope of removing unused headers, any specific reason to include it here?

Reply:
Oh good catch! I've added to make it compile on my laptop and forgot to remove.

Reply:
Addressed in 9384ccab82bd80c33823e46e060178b1a22740c3